src/m365/planner/commands/plan/plan-set.ts

```diff
@@ -0,0 +1,156 @@
+import { Cli, CommandOutput, Logger } from '../../../../cli';
+import Command, {
+  CommandOption,
+  CommandError
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import request from '../../../../request';
+import { Options as PlanGetCommandOptions } from '../plan/plan-get';
+import * as planGetCommand from '../plan/plan-get';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  title?: string;
+  newTitle: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanSetCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return commands.PLAN_SET;
+  }
+
+  public get description(): string {
+    return 'Set a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (error?: any) => void): void {
+    const getPlan = (): Promise<CommandOutput> => {
+      if (this.verbose) {
+        logger.logToStderr(`Retrieving plan ${args.options.id || args.options.title} ...`);
+      }
+
+      const options: PlanGetCommandOptions = {
+        ...args.options,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      return Cli.executeCommandWithOutput(planGetCommand as Command, { options: { ...options, _: [] } });
+    };
+
+    const updatePlan = (plan: any): Promise<void> => {
+      if (this.verbose) {
+        logger.logToStderr(`Updating plan with id ${plan['id']} ...`);
+      }
+
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/planner/plans/${plan['id']}`,
+        headers: {
+          'accept': 'application/json',
+          'If-Match': `${plan["@odata.etag"]}`
+        },
+        responseType: 'json',
+        data: {
+          title: args.options.newTitle
+        }
+      };
+
+      return request.patch(requestOptions);
+    };
+
+    getPlan()
+      .then((output: CommandOutput): void => {
+        if (!output.stdout) {
+          cb(new CommandError(`No plan found`));
+          return;
+        }
+        let plans: any = JSON.parse(output.stdout);
+        if(!Array.isArray(plans)) {
```

Comment:
Is there a reason why we're expecting an array here as we're calling `planner plan get` which returns 0 or 1 plans?

Reply:
Because `planner get` can give more than 1 plan back. It's possible to have multiple plans with the same name in the same group.

`m365 planner plan get --title Plan22 --ownerGroupName Testsite` --> returns 3 plans

![image](https://user-images.githubusercontent.com/36161889/144719136-0fe7870b-9b0d-4e20-9383-2c08508df986.png)

`if(!Array.isArray(plans)) ` --> Check if more than 1 plan is returned

Reply:
Understood, so that's rather unexpected and perhaps something we should rethink. All other get commands in the CLI return one object. If the underlying API is different, then we should add the necessary code to align the behavior with other get commands in the CLI and return just one plan.

Reply:
If we take the route of returning 1 plan, which one do we return? (last created?)
I do agree that it's highly unlikely that someone has 2 or more plans with the same name, but it's better to cover or at least think about this possible scenario. I don't know why they didn't build a restriction for creating a plan with the same name in the same group?

Reply:
In that case, we should return an error and ask user to disambiguate by specifying an ID (and in the error message we should show the list of retrieved IDs) rather than a name.

Reply:
![image](https://user-images.githubusercontent.com/36161889/146453424-ecc33f17-5105-4cd9-a8c7-c5f8c88ca29d.png)

Is this enough? Any suggestions ?

Reply:
This is good 👍